app/Console/Commands/ExifLens.php

```diff
@@ -41,7 +42,12 @@ public function handle(): int
 			$argument = $this->argument('nb');
 			$from = $this->argument('from');
 			$timeout = (int) $this->argument('tm');
-			\Safe\set_time_limit($timeout);
+
+			try {
+				\Safe\set_time_limit($timeout);
+			} catch (InfoException) {
+				// We do nothing, set_time_limit will throw an error on tests.
```

Comment:
```suggestion
				// We do nothing; set_time_limit will throw an error, if larger than `max_execution_time` in `php.ini` due to security concerns
```

Actually, this has nothing to do with testing, but with settings in `php.ini`. If `max_execution_time` is set to a non-zero value in `php.ini`, this value cannot be surpassed by `set_time_limit`. `set_time_limit` is more a mean to self-restrict the own runtime to a lower value or - if `max_execution_time` is set to zero in `php.ini` which means "infinity" - to restrict the runtime in the first place.

I think I already have written it somewhere else. Until know I haven't spend a thought on it, because the non-safe method simply silently failed, but maybe we should remove `set_time_limit` completely?

In this context, it makes sense, because it is actually serves as a mean of self-restriction for the CLI script and safety measure. I can imagine that there configurations which allow a PHP command line script to run indefinitely. But even here, we could be a little bit more nuanced. Something like this

```php
$maxAllowedTimeout = (int) ini_get('max_execution_time');
if ($timeout < maxAllowedTimeout || maxAllowedTimeout === 0) {
   \Safe\set_time_limit($timeout);
} else {
  $this->line('Cannot increase timeout beyond the configured system value . ' maxAllowedTimeout);
}
```

But we also use `set_time_limit` in `Import::fromUrl` and `Import::fromServer` and try to increase the time limit in every iteration. This does not make sense at all, because it simply does not work this way.

Any thoughts? @ildyria , @kamil4 ?

Reply:
I believe I may have added the `set_time_limit(ini_get('max_execution_time'));` calls. Their purpose is to _extend_ the execution time limit for operations that can use significant CPU time, such as an import from the server of a folder with many photos, or download of an album with many photos. As far as I understand, invoking `set_time_limit` restarts the timer from `0`, which is why we invoke it for every iteration.

I'm naively assuming that when I added this code, I tested that it actually works as intended, so needless to say I'm surprised by your statement that it "does not make sense at all, because it simply does not work this way". Have you actually tried commenting out those calls and seeing what happens when you try to import from the server a folder with, say, a hundred photos?

Reply:
Have left it for now this way, but I have fixed all invocation of `set_time_limit` consistently in https://github.com/LycheeOrg/Lychee/pull/1336/commits/67962c91318de2146bf22694873143df13b68baf.